compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/test/create_logical_objectfifo_link.mlir

```diff
@@ -0,0 +1,51 @@
+// RUN: iree-opt --split-input-file --pass-pipeline="builtin.module(func.func(iree-amdaie-create-logical-objectfifo-link))" --verify-diagnostics %s | FileCheck %s
+
+// CHECK-LABEL: func.func @link
+// CHECK:       %[[DMA0:.+]] = amdaie.circular_dma_cpy_nd
+// CHECK:       %[[DMA1:.+]] = amdaie.circular_dma_cpy_nd
+// CHECK:       amdaie.logicalobjectfifo.link
+// CHECK-SAME:  %[[DMA0]]
+// CHECK-SAME:  %[[DMA1]]
+func.func @link(%arg0: memref<32x1024xi32>, %arg1: memref<32x64xi32, 1>, %arg2: memref<8x8x4x8xi32, 2>) {
+  %0 = amdaie.logicalobjectfifo.from_memref %arg0, {} :memref<32x1024xi32> -> !amdaie.logicalobjectfifo<memref<32x1024xi32>>
+  %1 = amdaie.logicalobjectfifo.from_memref %arg1, {} : memref<32x64xi32, 1> -> !amdaie.logicalobjectfifo<memref<32x64xi32, 1>>
+  %2 = amdaie.logicalobjectfifo.from_memref %arg2, {} : memref<8x8x4x8xi32, 2> -> !amdaie.logicalobjectfifo<memref<8x8x4x8xi32, 2>>
+  %3 = amdaie.circular_dma_cpy_nd(%1[] [] [], %0[] [] []) : (!amdaie.logicalobjectfifo<memref<32x64xi32, 1>>, !amdaie.logicalobjectfifo<memref<32x1024xi32>>)
+  %4 = amdaie.circular_dma_cpy_nd(%2[] [] [], %1[] [] []) : (!amdaie.logicalobjectfifo<memref<8x8x4x8xi32, 2>>, !amdaie.logicalobjectfifo<memref<32x64xi32, 1>>)
+  return
+}
+
+// -----
+
+// CHECK-LABEL: func.func @link_two_inputs
+// CHECK:       %[[DMA0:.+]] = amdaie.circular_dma_cpy_nd
+// CHECK:       %[[DMA1:.+]] = amdaie.circular_dma_cpy_nd
+// CHECK:       %[[DMA2:.+]] = amdaie.circular_dma_cpy_nd
+// CHECK:       amdaie.logicalobjectfifo.link
+// CHECK-DAG:   %[[DMA0]]
+// CHECK-DAG:   %[[DMA1]]
+// CHECK-DAG:   %[[DMA2]]
+func.func @link_two_inputs(%arg0: memref<32x1024xi32>, %arg1: memref<32x64xi32, 1>, %arg2: memref<8x8x4x8xi32, 2>) {
```

Comment:
Is there also a case linking a input with two outputs?

Reply:
Good point, added it.